### PR TITLE
Implement Conditional Expression Operators

### LINKED ISSUE #84: Implement Conditional Expression Operators dsl
https://www.mongodb.com/docs/manual/reference/operator/aggregation/#conditional-expression-operators

### PR DESCRIPTION
**Pull request by masonJS**
resolves #84 

---
Hi, master. 
Among the codes I wrote, the `$switch` operator has not yet written the sample code.

I want to check the `$switch` operator code I designed first and proceed to the next.

### CODE REVIEW

File: core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/expression/conditional/SwitchExpressionDsl.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        branches.add(
+            ConditionalOperators.Switch.CaseOperator
+                .`when`(AggregationExpressionDsl().caseExpression())
+                .then(thenValue),
+        )
+    }
+
+    /**
+     * @param defaultValue Optional. The path to take if no branch case expression evaluates to true.
+     * Although optional, if default is unspecified and no branch case evaluates to true, $switch returns an error.
```
Comment (jbl428) [inline core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/expression/conditional/SwitchExpressionDsl.kt:36]: we can make this lines for function docs and param have simple description.

```suggestion
     * The path to take if no branch case expression evaluates to true.
     * Although optional, if default is unspecified and no branch case evaluates to true, $switch returns an error.
     * 
     *  @param defaultValue The default value.
```

**Comment by jbl428** [comment]
@haruInflearn I think you can make `$switch` example.

**Comment by masonJS** [comment]
> @haruInflearn I think you can make `$switch` example.

@jbl428 
i appreciate for your time.
so i finish this work.

**Review by jbl428 — APPROVED** [review]
LGTM

**Pull request merged**